### PR TITLE
Avoid unnecessary or duplicate PackageReferences - 7.0 Preview4 SDK work

### PR DESCRIPTION
**Pull request by ViktorHofer**
The 7.0 Preview4 SDK contains a recent NuGet change that warns when duplicate PackageReferences items are declared. Fixing the occurrences that I found offline.

### CODE REVIEW

**Review by carlossanlop — APPROVED** [review]

**Comment by carlossanlop** [comment]
Is the linker failure related?
```
Child node "4" exited prematurely. Shutting down. Diagnostic information may be found in files in "/tmp/" and will be named MSBuild_*.failure.txt. This location can be changed by setting the MSBUILDDEBUGPATH environment variable to a different directory.
```

**Comment by ViktorHofer** [comment]
Nope, that's an msbuild issue that we have seen in other PRs as well lately.

**Comment by ViktorHofer** [comment]
I had an earlier approval from Carlos, I think this is save to merge. Unblock devs who currently use a P4 SDK to restore, build and test runtime.

File: eng/testing/xunit/xunit.props [hunk 1]
```diff
@@ -12,18 +12,19 @@
...
 
-  <ItemGroup Condition="'$(ArchiveTests)' != 'true' and '$(PublishingTestsRun)' != 'true' and '$(TestSingleFile)' != 'true'">
-    <!-- Microsoft.Net.Test.Sdk brings a lot of assemblies with it. To reduce helix payload submission size we disable it on CI. -->
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="$(MicrosoftNETTestSdkVersion)" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="$(XUnitRunnerVisualStudioVersion)" GeneratePathProperty="true" />
     <!--
       Microsoft.Net.Test.Sdk has a dependency on Newtonsoft.Json v9.0.1. We upgrade the dependency version
       with the one used in libraries to have a consistent set of dependency versions. Additionally this works
       around a dupliate type between System.Runtime.Serialization.Formatters and Newtonsoft.Json.
     -->
```
Comment (am11) [inline eng/testing/xunit/xunit.props:20]: Is this comment still relevant? Going by the dependency graph over at https://www.nuget.org/packages/Microsoft.NET.Test.Sdk/, only UAP has dependency on Newtonsoft.Json. UAP support in libraries was dropped in https://github.com/dotnet/corefx/pull/41759.

File: eng/testing/xunit/xunit.props [hunk 2]
```diff
@@ -12,18 +12,19 @@
...
     <PackageReference Include="Newtonsoft.Json" Version="$(NewtonsoftJsonVersion)" />
```
Comment (am11) [inline eng/testing/xunit/xunit.props:21]: Some (non-shipping) test projects under installers have hardcoded version of Newtonsoft.Json https://grep.app/search?q=Newtonsoft.Json&case=true&filter[repo][0]=dotnet/runtime&filter[path.pattern][0]=src/installer&filter[lang][0]=XML Can we use `NewtonsoftJsonVersion` for them as well?

**Review by am11 — COMMENTED** [review]
Two questions/comemnts related to Newtonsoft.Json. :)

File: eng/testing/xunit/xunit.props (hunk 1 above)
Comment (ViktorHofer) [reply eng/testing/xunit/xunit.props:20]: The dependency is still there. Just click through the netcoreapp dependencies and you will find newtonsoft.json.

File: eng/testing/xunit/xunit.props (hunk 2 above)
Comment (ViktorHofer) [reply eng/testing/xunit/xunit.props:21]: Yes that dependency Version should be bumped or the dependency dropped entirely in favor of STJ

**Pull request merged**